### ISSUE TITLE
email provider not linking with existing facebook account. 

### ISSUE THREAD

**Issue by j2emanue** [firebase/FirebaseUI-Android#1275]
Regarding firebaseUI auth, i found that if an email account is already linked with facebook provider there is an error of 

```
W/AccountLinker: Error linking with credential
    com.google.firebase.FirebaseException: User has already been linked to the given provider.
        at com.google.firebase.auth.api.internal.zzce.zzb(Unknown Source:221)
        at com.google.firebase.auth.api.internal.zzao.zzd(Unknown Source:35)
        at com.google.firebase.auth.FirebaseAuth.zzc(Unknown Source:15)
        at com.google.firebase.auth.FirebaseUser.linkWithCredential(Unknown Source:11)
        at com.firebase.ui.auth.util.accountlink.AccountLinker.start(AccountLinker.java:82)
        at com.firebase.ui.auth.util.accountlink.AccountLinker.<init>(AccountLinker.java:52)
        at com.firebase.ui.auth.util.accountlink.AccountLinker.linkToNewUser(AccountLinker.java:65)
        at com.firebase.ui.auth.viewmodel.idp.LinkingSocialProviderResponseHandler.startSignIn(LinkingSocialProviderResponseHandler.java:43)
        at com.firebase.ui.auth.ui.idp.WelcomeBackIdpPrompt$1.onSuccess(WelcomeBackIdpPrompt.java:129)
        at com.firebase.ui.auth.ui.idp.WelcomeBackIdpPrompt$1.onSuccess(WelcomeBackIdpPrompt.java:126)
        at com.firebase.ui.auth.viewmodel.ResourceObserver.onChanged(ResourceObserver.java:45)
        at com.firebase.ui.auth.viewmodel.ResourceObserver.onChanged(ResourceObserver.java:14)
        at android.arch.lifecycle.LiveData.considerNotify(LiveData.java:109)
        at android.arch.lifecycle.LiveData.dispatchingValue(LiveData.java:126)
        at android.arch.lifecycle.LiveData.setValue(LiveData.java:282)
        at android.arch.lifecycle.MutableLiveData.setValue(MutableLiveData.java:33)
        at com.firebase.ui.auth.viewmodel.OperableViewModel.setResult(OperableViewModel.java:24)
        at com.firebase.ui.auth.data.remote.FacebookSignInHandler.access$600(FacebookSignInHandler.java:39)
        at com.firebase.ui.auth.data.remote.FacebookSignInHandler$ProfileRequest.onCompleted(FacebookSignInHandler.java:172)
        at com.facebook.GraphRequest$1.onCompleted(GraphRequest.java:301)
        at com.facebook.GraphRequest$5.run(GraphRequest.java:1380)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
```

and that error is probably fine. but there  NO CALBACK to onActivityResults. instead i get a toast that an error has occured.  

Here are the steps to reproduce:

1. login with facebook

2. logout

3. signin with email provider and use the same email as the facebook account.

3b. user then get the prompt: you've already used facebookEmail@domain.com . Sign in with facebook to continue

4. after facebook prompt appears to resolve the collision enter in facebook credentials (or hit continue if it does not prompt etc)

5. library gives the above error and a toast message that an error has occured. 

expected results: i get a callback with result OK and move forward. 

for your info in the firebase console the email account i tested with only had the facebook provider listed.

**Comment by j2emanue** [comment]
dont close this yet. let me try again to give you steps. i'll grab the updates that you guys up in past few days also. will report back soon.

**Comment by j2emanue** [comment]
do you guys recommend i fork this repo or Alex's if i need the most recent changes ?

**Comment by samtstern** [comment]
What changes are you looking for specifically?

**Comment by SUPERCILEX** [comment]
There are actually [snapshot builds](https://github.com/firebase/FirebaseUI-Android/blob/master/README.md#snapshot-builds) now, so unless you need the advanced account linking, those should cover it. 👍

**Comment by SUPERCILEX** [comment]
Oh wait, you filed https://github.com/SUPERCILEX/FirebaseUI-Android/issues/27! So actually, this turns out to be a major oversight on my part that primarily affects my fork, but would have nipped us in the butt pretty bad later on had you not caught it. So... thanks! ❤️ I've already included a fix in `f0a391a50b` of my fork and you can view the PR: #1293.

**Comment by samtstern** [comment]
#1293 is merged so there should be a 4.0.0 SNAPSHOT build availabile with the fix in a half hour or less.  Would love if you could confirm that it fixes your problem.

**Comment by j2emanue** [comment]
i already put the changes in manually since it was a not big changes. its working now.
